### DISCUSSION TITLE
posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models/

### DISCUSSION THREAD

**Discussion by giscus** [mvuorre/mvuorre.github.io#15] [giscus]
# posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models/

A reasonable choice might be the zero-one-inflated beta model

https://mvuorre.github.io/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models/

**Comment by qdread** [comment]
Thank you for this excellent tutorial! I was wondering if you know of any implementation in R of a function like `dzoib()` that would allow you to plot a probability density function for a zero-one-inflated beta distribution. I am trying to implement one myself but running into issues.

**Reply by mvuorre** [reply]
Hi, sorry I don't know any off the top of my head.

**Comment by tanhuizhen** [comment]
Thank you for this amazing write up Matti! It was a fantastic introduction to running zoib regression and interpretation of the results. May I know if you will be releasing the rzoib() function?

**Reply by mvuorre** [reply]
https://github.com/mvuorre/mvuorre.github.io/blob/18aec45d1145f4383b0646b19b74e9ae18041344/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models/index.qmd#L63 :)

**Reply by tanhuizhen** [reply]
Thank you again!!

**Reply by mvuorre** [reply]
Thanks 😊

**Comment by kjwalter89** [comment]
This is such a great tutorial which really clarified so much for me! I am wondering how one might generate predictions for the joint 0 and beta part of the model separately from the coi part of the model. In other words, I understand how one would generate the predicted probabilities for the 1s (coi), for the beta distribution (mu), and the 0s (zoi*(1-coi), I believe?). Is it possible to get the joint predictions of the 0's and beta, perhaps something like mu*(1-(zoi*(1-coi))). Any insight into this would be much appreciated.

**Reply by mvuorre** [reply]
You should get all components of the outcome model using standard brms prediction functions. If you don't mean that I'm not sure... Maybe you can post a reproducible example with a clear pointer as to what you want to achieve somewhere? (I'd recommend the Stan discourse [https://discourse.mc-stan.org/], feel free to tag me [at]matti.)

**Comment by matti** [comment]
I believe that @matti is wrong @mvuorre

**Reply by mvuorre** [reply]
Sorry, I tried to escape the @ with a \ but apparently GH didn't like it. 🤦 Apologies.

**Comment by peter1328** [comment]
Hi Matti!
I re-visited this blog post after now understanding the ZOIB myself a bit :D
I am wondering whether your simulation makes sense. Usually any variable that affects the distribution in the beta range will affect zeros and ones too. For example, an intervention that increases something will likely affect those in the 0s going up a bit just as those reaching the maximum actually reaching it, instead of effects magically going only between 0.1 and 99.9. And in those cases, using a t-test will very likely point to effects rather reliably. So I am not unconvinced and will examine potential differences in my own data before trusting that ZOIB is some magical tool for slider data. Of course, I do believe that it has potential, though, and in particular, if we really want our model to fit our data, or when effects of interest may take place close to the 0s or 1s and the overall distribution has a lot of density close to these borders, then it is a great tool.

Cheers!
Peter

**Reply by mvuorre** [reply]
Hi, yes ZOIB is certainly not a magical tool. Check out ordered beta or e.g. censored normal models for something more in the ballpark of what you're talking about.
